pkg/handlers/generic/lifecycle/serviceloadbalancer/metallb/handler.go

```diff
@@ -160,6 +162,21 @@ func (n *MetalLB) Apply(
 						},
 					},
 				); err != nil {
+					if apierrors.IsConflict(err) {
+						switch o.GetKind() {
+						case "IPAddressPool":
+							err = fmt.Errorf(
+								"%w. This resource must list only the addresses listed in the Cluster configuration",
```

Comment:
nit: Does "must list only the addresses listed" sound a little strange to you?
```suggestion
								"%w. This resource has bee modified, it must contain exactly the addresses listed in the Cluster configuration",
```